activerecord/test/cases/schema_loading_test.rb

```diff
@@ -42,6 +42,19 @@ def test_model_with_changed_custom_lock_is_loaded_twice
     assert_equal 2, klass.load_schema_calls
   end
 
+  def test_load_with_schema_cache_without_connecting
+    model_one = define_model
+    model_one.define_attribute_methods
+
+    model_two = define_model
+    assert_not_nil model_two.connection_pool.schema_cache
+    model_two.connection_pool.schema_cache.stub(:connection, :no_connection) do
```

Comment:
These stubs are super ugly, I'd like to find a good test helper to simulate a down database.